src/pages/inward-transfer/forms/InwardTransferForm.js

```diff
@@ -0,0 +1,1093 @@
+    notes: '',
+    amount: null,
+    transferType: null,
+    faxNo: '',
+    sender_firstName: '',
+    sender_lastName: '',
+    sender_middleName: '',
+    sender_nationalityId: null,
+    sender_phone: null,
+    sender_otherInfo: '',
+    sender_countryId: null,
+    sender_idtId: null,
+    sender_idNo: '',
+    sender_idIssuePlace: '',
+    sender_idIssueDate: null,
+    sender_idExpiryDate: null,
+    receiver_type: '',
+    receiver_riskCategory: '',
+    receiver_payoutType: '',
+    receiver_isResident: false,
+    receiver_firstName: '',
+    receiver_middleName: '',
+    receiver_lastName: '',
+    receiver_fl_firstName: '',
+    receiver_fl_middleName: '',
+    receiver_fl_lastName: '',
+    receiver_phone: null,
+    receiver_nationalityId: null,
+    receiver_idtId: null,
+    receiver_idNo: '',
+    receiver_idIssueDate: null,
+    receiver_idExpiryDate: null,
+    receiver_idIssuePlace: null,
+    receiver_accountNo: '',
+    receiver_address1: '',
+    receiver_address2: '',
+    receiver_bank: '',
+    receiver_bankBranch: '',
+    receiver_ttNo: '',
+    paymentMode: '',
+    paymentBank: '',
+    commissionType: '',
+    commissionAgent: '',
+    commissionReceiver: '',
+    expiryDate: null,
+    sourceOfIncome: '',
+    purposeOfTransfer: ''
+  })
+
+  const { maxAccess } = useDocumentType({
+    functionId: SystemFunction.InwardTransfer,
+    access: access,
+    enabled: !recordId
+  })
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      date: yup.date().required(),
+      corId: yup.string().required(),
+      currencyId: yup.string().required(),
+      amount: yup.number().required(),
+      transferType: yup.string().required(),
+      faxNo: transferType == '1' ? yup.string().required() : yup.string().notRequired(),
+      sender_firstName: yup.string().required(),
+      sender_lastName: yup.string().required(),
+      sender_nationalityId: yup.string().required(),
+      sender_countryId: yup.string().required(),
+      receiver_type: yup.string().required(),
+      receiver_firstName: yup.string().required(),
+      receiver_lastName: yup.string().required(),
+      receiver_payoutType: yup.string().required(),
+      commissionAgent: yup.number().required(),
+      commissionReceiver: yup.number().required()
+    }),
+    onSubmit: async values => {
+      const copy = { ...formik.values }
+      copy.date = formatDateToApi(copy?.date)
+      copy.wip = copy?.wip === '' ? 1 : copy?.wip
+      copy.exRate = copy?.exRate === '' ? 1 : copy?.exRate
+      copy.status = copy?.status === null ? 1 : copy?.status
+      copy.baseAmount = copy?.baseAmount === '' ? copy?.amount : copy?.baseAmount
+      copy.rateCalcMethod = copy?.rateCalcMethod === '' ? 1 : copy?.rateCalcMethod
+      copy.sender_idIssueDate = copy.sender_idIssueDate ? formatDateToApi(copy?.sender_idIssueDate) : null
```

Comment:
copy.sender_idIssueDate = copy.sender_idIssueDate ? formatDateToApi(copy?.sender_idIssueDate) : null
      copy.sender_idExpiryDate = copy.sender_idExpiryDate ? formatDateToApi(copy?.sender_idExpiryDate) : null
      copy.receiver_idIssueDate = copy.receiver_idIssueDate ? formatDateToApi(copy?.receiver_idIssueDate) : null
      copy.receiver_idExpiryDate = copy.receiver_idExpiryDate ? formatDateToApi(copy?.receiver_idExpiryDate) : null
      copy.expiryDate = copy.expiryDate ? formatDateToApi(copy?.expiryDate) : null
      
   How is the logic?
   if they are null or undefined how we are formatting them?